docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,338 @@
+|250
+
+|`microprofile-config.properties` from application
+|100
+|===
+
+In this case, `my.prop` from `InMemoryConfigSource` will only be used if the Config Engine is unable to find a value
+in `System Properties`, `Environment Variables from System` and `Environment Variables from .env file` in this order.
+
+[[config-source-factory]]
+== Config Source Factory
+
+Another way to create a `ConfigSource` is via the https://github.com/smallrye/smallrye-config[SmallRye Config]
+`io.smallrye.config.ConfigSourceFactory`. The difference between the SmallRye Config Factory and the standard way to
+create a `ConfigSource` as specified in MicroProfile Config, is the Factory ability to provide a context with access to
+the available configuration.
+
+Registration of a `io.smallrye.config.ConfigSourceFactory` is done via the ServiceLoader mechanism by providing the
+implementation classes in a `META-INF/services/io.smallrye.config.ConfigSourceFactory file`.
+
+=== Example
+
+Consider:
+
+.org.acme.config.URLConfigSourceFactory
+[source,java]
+----
+package org.acme.config;
+
+import java.util.Collections;
+import java.util.OptionalInt;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import io.smallrye.config.ConfigSourceContext;
+import io.smallrye.config.ConfigSourceFactory;
+import io.smallrye.config.ConfigValue;
+import io.smallrye.config.PropertiesConfigSource;
+
+public class URLConfigSourceFactory implements ConfigSourceFactory {
+    @Override
+    public Iterable<ConfigSource> getConfigSources(final ConfigSourceContext context) {
+        final ConfigValue value = context.getValue("config.url");
+        if (value == null || value.getValue() == null) {
+            return Collections.emptyList();
+        }
+
+        try {
+            return Collections.singletonList(new PropertiesConfigSource(new URL(value.getValue())));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public OptionalInt getPriority() {
+        return OptionalInt.of(290);
+    }
+}
+----
+
+And registration in:
+
+.META-INF/services/io.smallrye.config.ConfigSourceFactory
+[source,properties]
+----
+org.acme.config.URLConfigSourceFactory
+----
+
+By implementing `io.smallrye.config.ConfigSourceFactory`, a list of `ConfigSource` may be provided via the
+`Iterable<ConfigSource> getConfigSources(ConfigSourceContext context)` method. The `ConfigSourceFactory` may also
+assign a priority by overriding the default method `OptionalInt getPriority()`. This is only used to sort multiple
+factories during initialization. Once all the factories are initialized, the provided list of `ConfigSource` will use
+their own ordinal and sorted with all the other sources available in the `Config` instance.
+
+When the Factory is initializing, the provided `ConfigSourceContext` may call the method
+`ConfigValue getValue(String name)`. This method lookups configuration names in all `ConfigSource` that were already
```

Comment:
"lookups" is not a verb.

Should be "looks up".

Also, "in all the `ConfigSource`s" would read much better than "in all `ConfigSource`", it seems to me.